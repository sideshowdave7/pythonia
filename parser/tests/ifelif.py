id = 5
if (id == 1 or id not in [1, 2, 3] and not id is not 5 and not id is 5):
  print("1")
elif (id >> 4 > 0 and -id > 0 and +id < 0 and ~id - -id + +id == 0):
  print("2")
elif (id ** 2 % 3 // 2 == 9):
  print("3")
else:
  print("Good")

