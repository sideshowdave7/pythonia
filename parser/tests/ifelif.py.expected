(program
 (= (id) 5)
 (cond
  ((or (comparison id ("==" 1))
       (and (comparison id ("not-in" (list 1 2 3)))
            (not (comparison id ("is-not" 5)))
            (not (comparison id ("is" 5)))))
   (suite (expr (indexed print (called "1")))))
  ((and (comparison (shift id (">>" 4)) (">" 0))
        (comparison ("-" id) (">" 0))
        (comparison ("+" id) ("<" 0))
        (comparison (arith ("~" id) ("-" ("-" id)) ("+" ("+" id))) ("==" 0)))
   (suite (expr (indexed print (called "2")))))
  ((comparison (term (power id 2) ("%" 3) ("//" 2)) ("==" 9))
   (suite (expr (indexed print (called "3")))))
  (else (suite (expr (indexed print (called "Good")))))))

